File: Data-Agent-Server/data-agent-server-app/src/main/java/edu/zsc/ai/domain/service/db/TableDataService.java
package edu.zsc.ai.domain.service.db;

import edu.zsc.ai.domain.model.dto.response.db.TableDataResponse;

public interface TableDataService {

    /**
     * Get table data with pagination
     *
     * @param connectionId connection ID
     * @param catalog catalog name
     * @param schema schema name
     * @param tableName table or view name
     * @param userId user ID
     * @param currentPage current page number
     * @param pageSize page size
     * @return table data response
     */
    TableDataResponse getTableData(Long connectionId, String catalog, String schema, String tableName, Long userId, Integer currentPage, Integer pageSize);

    /**
     * Get view data with pagination
     *
     * @param connectionId connection ID
     * @param catalog catalog name
     * @param schema schema name
     * @param viewName view name
     * @param userId user ID
     * @param currentPage current page number
     * @param pageSize page size
     * @return table data response
     */
    TableDataResponse getViewData(Long connectionId, String catalog, String schema, String viewName, Long userId, Integer currentPage, Integer pageSize);
}

File: Data-Agent-Client/src/services/tableData.service.ts
import http from '../lib/http';
import { ApiPaths } from '../constants/apiPaths';

export interface TableDataResponse {
  headers: string[];
  rows: unknown[][];
  totalCount: number;
  currentPage: number;
  pageSize: number;
  totalPages: number;
}

export const tableDataService = {
  getTableData: async (
    connectionId: string,
    tableName: string,
    catalog?: string,
    schema?: string,
    currentPage: number = 1,
    pageSize: number = 100
  ): Promise<TableDataResponse> => {
    const params: Record<string, string | number> = {
      connectionId,
      tableName,
      currentPage,
      pageSize
    };
    if (catalog != null && catalog !== '') params.catalog = catalog;
    if (schema != null && schema !== '') params.schema = schema;

    const response = await http.get<TableDataResponse>(ApiPaths.TABLE_DATA, { params });
    return response.data;
  },

  getViewData: async (
    connectionId: string,
    viewName: string,
    catalog?: string,
    schema?: string,
    currentPage: number = 1,
    pageSize: number = 100
  ): Promise<TableDataResponse> => {
    const params: Record<string, string | number> = {
      connectionId,
      viewName,
      currentPage,
      pageSize
    };
    if (catalog != null && catalog !== '') params.catalog = catalog;
    if (schema != null && schema !== '') params.schema = schema;

    const response = await http.get<TableDataResponse>(ApiPaths.VIEW_DATA, { params });
    return response.data;
  },
};
